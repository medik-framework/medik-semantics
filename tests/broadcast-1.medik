machine Broadcaster {
  var subscriber;
  init state Start {
    entry(subscriber) {
      this.subscriber = subscriber;
      var i = 0;
      while(i < 5) {
        broadcast eTick;
        sleep(1);
        i = i + 1;
      }
    }
  }
}

init machine Receiver
  recieves eTick {
    init state Start {
      entry {
        new Broadcaster(this);
        goto Recieve;
      }
    }

    state Recieve {
      on eTick do {
        print("Recieved Tick\n");
        goto Recieve;
      }
    }
}

