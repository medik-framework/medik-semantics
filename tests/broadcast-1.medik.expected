Recieved Tick
Recieved Tick
Recieved Tick
Recieved Tick
Recieved Tick
